python/ray/experimental/serve/queues.py

```diff
@@ -77,27 +103,38 @@ class CentralizedQueues:
     """
 
     def __init__(self):
+        # Note: Several queues are used in the router
+        # - When a request come in, it's placed inside its corresponding
+        #   service_queue.
+        # - The service_queue is dequed during flush operation, which moves
+        #   the queries to backend buffer_queue. Here we match a request
+        #   for a service to a backend given some policy.
+        # - The worker_queue is used to collect idle actor handle. These
+        #   handles are dequed during the second stage of flush operation,
+        #   which assign queries in buffer_queue to actor handle.
+
+        # -- Queues -- #
+
         # service_name -> request queue
-        self.queues = defaultdict(deque)
+        self.service_queues: DefaultDict[asyncio.Queue[Query]] = defaultdict(
+            asyncio.Queue)
+        # backend_name -> worker request queue
+        self.worker_queues: DefaultDict[asyncio.Queue[
+            ray.actor.ActorHandle]] = defaultdict(asyncio.Queue)
+        # backend_name -> worker payload queue
+        self.buffer_queues = defaultdict(blist.sortedlist)
+
+        # -- Metadata -- #
 
         # service_name -> traffic_policy
         self.traffic = defaultdict(dict)
-
         # backend_name -> backend_config
         self.backend_info = dict()
 
-        # backend_name -> worker request queue
-        self.workers = defaultdict(deque)
+        # -- Synchronization -- #
 
-        # backend_name -> worker payload queue
-        # using blist sortedlist for deadline awareness
-        # blist is chosen because:
-        # 1. pop operation should be O(1) (amortized)
-        #    (helpful even for batched pop)
-        # 2. There should not be significant overhead in
-        #    maintaining the sorted list.
-        # 3. The blist implementation is fast and uses C extensions.
-        self.buffer_queues = defaultdict(sortedlist)
+        # Only one flush operation can happen at a time
```

Comment:
Can you add more information why one flush operation needed at a time? As in what is the problem with multiple flush operation running concurrently?